Release/v6.6.1

kttkjl opened the pull request:
IPR Client v6.6.1 Release Notes

Bugfixes:
- [DEVSU-1683] - Fixed CivicCellRenderer to take into account when externalSource is of type array
- [DEVSU-1686] - Added tpm and kIQR for SM, CNV, SV

Commit:
fix: update civicCellRenderer to take into account when externalSource are of type array

DEVSU-1683

Commit:
6.6.1

Anna-log7 commented on app/components/DataTable/components/CivicCellRenderer/index.tsx:
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
+  const {
+    externalStatementId,
+    externalSource,
+  } = data;
```
You can move this destructure into the props like:

```
const CivicCellRenderer = ({
  data: { externalStatementId, externalSource },
}: CivicCellRendererProps): JSX.Element => {
```

kttkjl replied:
I could, but this I feel like it's cleaner?

Anna-log7 replied:
Up to you then if you feel that way

kttkjl commented on app/components/DataTable/components/CivicCellRenderer/index.tsx:
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
-    } else {
-      setText(data.externalSource);
+    if (data) {
+      if (Array.isArray(externalSource)) {
```
good catch, thanks!

kttkjl commented on app/components/DataTable/components/CivicCellRenderer/index.tsx:
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
+    if (data) {
+      if (Array.isArray(externalSource)) {
+        const numOnly = /^\d+$/;
+        if (externalSource.map((es) => es.toLowerCase()).includes('civic')) {
```
I thought externalSource was of type string, when would it be boolean?

Anna-log7 replied:
Argh I read this wrong sorry. The parentheses got me

Commit:
test: add story for civicCellRenderer

DEVSU-1683

Commit:
fix: add check for when source is civic and has multiple ids

DEVSU-1683

kttkjl commented on app/components/DataTable/components/CivicCellRenderer/index.tsx:
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
+    </MenuItem>
+  ));
+
+  if (links.length > 1) {
```
meaning we'll have to show their IDs too? 
Previously it was just setting the text of the external sources in the cell

there are stories for these situations now on storybook

Anna-log7 replied:
Nope, you've covered all the cases it looks like!

Anna-log7 commented:
Only other change I'd say is to make the links the same sort of style (both blue with underline). I've been only using icons in the actions column and I think it makes sense to stick with that for now.

![image](https://user-images.githubusercontent.com/51016800/142261703-17d37072-80c8-4f45-a97e-c98c41dbe3a8.png)

Anna-log7 commented on app/components/DataTable/components/CivicCellRenderer/index.tsx:
```diff
@@ -3,33 +3,94 @@ import React, { useState, useEffect } from 'react';
+  } = data;
+
   const [link, setLink] = useState('');
+  const [links, setLinks] = useState([]);
```
Should add a type here

Anna-log7 commented on app/components/DataTable/components/CivicCellRenderer/index.tsx:
```diff
@@ -3,33 +3,94 @@ import React, { useState, useEffect } from 'react';
   const [link, setLink] = useState('');
+  const [links, setLinks] = useState([]);
   const [text, setText] = useState('');
+  const [anchorEl, setAnchorEl] = useState(null);
```
Same for here with the type. You can type the event too in the handler below

Commit:
PR address: add types

Commit:
fix: apply changes done to Expression to SmallMut, CNV, SV
- Add colDef for tpm and kIQR for SM, CNV, SV
- Add interactions to show either rpkm/tpm

DEVSU-1686

Commit:
fix: add more consistent styling to CIViC links

DEVSU-1683

kttkjl commented:
> Only other change I'd say is to make the links the same sort of style (both blue with underline). I've been only using icons in the actions column and I think it makes sense to stick with that for now.
> 
> ![image](https://user-images.githubusercontent.com/51016800/142261703-17d37072-80c8-4f45-a97e-c98c41dbe3a8.png)

![image](https://user-images.githubusercontent.com/25511396/142357042-3b42ab76-d7b3-40db-b67f-21552202ec0f.png)

The top is when civic is array
The bottom is when civic is just a single value

Commit:
fix: add column hide to kIQR and TPM on SM, SV, and CNV

DEVSU-1686

Commit:
fix: condensed visible column logic for CNV, SM, SV

DEVSU-1686

kttkjl committed:
Merge pull request #122 from bcgsc/bugfix/DEVSU-1686-add-tpm-kIQR

fix: apply changes done to Expression to SmallMut, CNV, SV
